automation/patch_igz/patch_remote.py

```diff
@@ -34,97 +33,123 @@
         patch_yaml_data = yaml.safe_load(patch_file)
         self._deploy_patch = json.dumps(patch_yaml_data)
         self._reset_db = reset_db
-        self._tag = tag
-        self._patch_log_collector = bool(log_collector)
+        self._image_tag = image_tag
+        self._patch_log_collector_image = bool(patch_log_collector_image)
         self._validate_config()
-
-        nodes = self._config["DATA_NODES"]
-        if not isinstance(nodes, list):
-            nodes = [nodes]
-        self._nodes = nodes
-
-    def patch_mlrun_api(self):
+        self._patch_mlrun_image = patch_mlrun_image
+        self._skip_patch_api = skip_patch_api
+
+        cluster_data_nodes = self._config["DATA_NODES"]
+        if not isinstance(cluster_data_nodes, list):
+            cluster_data_nodes = [cluster_data_nodes]
+        self._cluster_data_nodes = cluster_data_nodes
+
+    def patch(self):
+        version = self._get_current_version()
+        image_tag = self._get_image_tag(version)
+        targets = []
+        if not self._skip_patch_api:
+            targets.append(Consts.api)
+        if self._patch_mlrun_image:
+            targets.append(Consts.mlrun)
+        if self._patch_log_collector_image:
+            targets.append(Consts.log_collector)
+        if not targets:
+            raise ValueError("No targets to patch")
         self._docker_login_if_configured()
-        vers = self._get_current_version()
 
-        image_tag = self._get_image_tag(vers)
-        built_api_image = self._make_mlrun("api", image_tag, "mlrun-api")
-
-        built_images = [built_api_image]
-        built_log_collector_image = None
-        if self._patch_log_collector:
-            built_log_collector_image = self._make_mlrun(
-                "log-collector", image_tag, "log-collector"
-            )
-            built_images.append(built_log_collector_image)
-
-        built_images = self._tag_images_for_multi_node_registries(built_images)
+        target_to_built_images = self._make_targets(
+            targets=targets,
+            image_tag=image_tag,
+        )
+        # Docker login and push images
+        built_images = self._tag_images_for_multi_node_registries(
+            target_to_built_images.values()
+        )
         self._push_docker_images(built_images)
 
-        node = self._nodes[0]
+        # Connect to the first node and start deployment patching process
+        node = self._cluster_data_nodes[0]
         self._connect_to_node(node)
-        try:
-            self._replace_deploy_policy()
-            self._replace_deployment_images(self.Consts.api_container, built_api_image)
-            if self._patch_log_collector:
-                # sanity
-                if not built_log_collector_image:
-                    raise RuntimeError("Log collector image not built")
+        if not self._skip_patch_api:
+            try:
+                # Replace deployment policies and images
+                self._replace_deploy_policy()
```

Comment:
please change the name it's not very indicative 
```suggestion
                self._patch_deployment_from_file()
```